leader-election-impl/src/test/java/com/palantir/paxos/PaxosConsensusFastTest.java

```diff
@@ -189,5 +186,67 @@ public void learnerRecovery() {
         cache.clear();
         state.gainLeadership(0);
     }
+
+    @Test
+    public void loseLeadershipAfterSteppingDown() {
+        LeadershipToken token = state.gainLeadership(0);
+        assertThat(state.leader(0).isStillLeading(token)).isEqualTo(StillLeadingStatus.LEADING);
+        assertThat(state.leader(0).stepDown()).isTrue();
+        assertThat(state.leader(0).isStillLeading(token)).isEqualTo(StillLeadingStatus.NOT_LEADING);
+    }
+
+    @Test
+    public void otherNodeCanBecomeLeaderAfterSteppingDown() {
+        LeadershipToken token = state.gainLeadership(0);
+        assertThat(state.leader(0).isStillLeading(token)).isEqualTo(StillLeadingStatus.LEADING);
+        assertThat(state.leader(0).stepDown()).isTrue();
+
+        LeadershipToken token2 = state.gainLeadership(1);
+        assertThat(state.leader(1).isStillLeading(token2)).isEqualTo(StillLeadingStatus.LEADING);
```

Comment:
is it worth double checking that `leader(0)` is *not* leading?

Reply:
Yep, can do - I thought this would be covered by the loseLeadershipAfterSteppingDown test, but fair enough - maybe something that makes 1 gain leadership also restores it for everyone